src/countries/countries.service.ts

```diff
@@ -1,4 +1,46 @@
 import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Country } from './country.entity';
+import { DeleteResult, Repository, UpdateResult } from 'typeorm';
+import { CountryInterface } from './country.interface';
 
 @Injectable()
-export class CountriesService {}
+export class CountriesService {
+  constructor(
+    @InjectRepository(Country)
+    private countryRepository: Repository<CountryInterface>,
+  ) {}
+
+  findAll(): Promise<CountryInterface[]> {
+    return this.countryRepository.find();
+  }
+
+  findOne(id: number): Promise<CountryInterface | null> {
+    return this.countryRepository.findOneById(id);
+  }
+
+  create(country: CountryInterface): Promise<CountryInterface> {
+    return this.countryRepository.save(this.countryRepository.create(country));
+  }
+
+  update(id: number, country: CountryInterface): Promise<UpdateResult> {
+    return this.countryRepository
+      .createQueryBuilder()
+      .update()
+      .set({
+        name: country.name,
+        logo: country.logo,
+      })
+      .where('id = :id', { id })
+      .execute();
+  }
+
+  remove(id: number): Promise<DeleteResult> {
```

Comment:
Again might be a personal opinion, but I prefer to never return to the user the result of a query (update result or delete result, etc...).

We settle the promise and depending on the outcome we either return a DTO or nothing in case of delete or we throw a custom error